.github/workflows/s3-bucket.yml

```diff
@@ -0,0 +1,44 @@
+name: S3 Bucket Test
+
+on:
+  push:
+  workflow_dispatch:
+
+jobs:
+  test-on-os-node-matrix:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        node: [18, 20, 22]
```

Comment:
Please use this matrix.
https://github.com/mapbox/node-pre-gyp/blob/a2785e88e6fec291b0df015da078deb0d00ba1ba/.github/workflows/ci.yml#L12-L16

This repo has been undermaintained for long periods, so it is better to signal early if there are incompatibilities with a new LTS.